tests/foreman/api/test_http_proxy.py

```diff
@@ -347,3 +349,62 @@ def _finalize():
     assert response.get('errors') is None
     assert repo.read().last_sync is not None
     assert repo.read().content_counts['rpm'] >= 1
+
+
+def test_refresh_updates_remotes_proxy(module_target_sat, module_org, module_product):
+    """Ensure that repo refresh updates the http-proxy of pulp remote.
+
+    :id: ab0f3734-40bd-4524-b8b9-0ab857ca3c3f
+
+    :setup:
+        1. Disable pulp CLI safe mode.
+        2. Create a product, repo and HTTP proxy.
+
+    :steps:
+        1. Get repository remote's href.
+        2. Ensure proxy_url matches the one from setup.
+        3. Set fake value to proxy_url, ensure it's written.
+        4. Refresh repos via rake and ensure the original proxy_url has been restored.
+
+    :expectedresults:
+        1. Repo refresh restores remote's original proxy_url value.
+
+    :verifies: SAT-26741
+
+    :customerscenario: true
+    """
+    sat = module_target_sat
+    sat.set_pulp_cli_safemode(False)
```

Comment:
Should we set this back to `True` when we finish the test? This seems like we are leaking state.

Reply:
I definitely would prefer a session scoped autouse fixture, I find that type of state management to be the most intuitive when reading the test, and the least cluttering

Reply:
Wouldn't it be optimal to apply such configurations at satlab level during checkouts? 🤔

Reply:
Also, after looking into [docs](https://pulpproject.org/pulp-cli/docs/user/reference/using_the_cli/), it appears, you should be able to execute a pulp command of your choice with `--dry-run` cli arg and thus, it shouldn't be necessary to touch the global config at all.

Reply:
I don't really know, it might be. We can ask @bherrin3 or @rdrazny for their opinions too. From functional POV it doesn't make much difference. 

I may dodge to `--dry-tun` (thanks for the tip!) it I feel too blocked by many contradictory opinions here. But still, if we are going to use pulp cli more often (in modify mode), global settings would be preferred.

Reply:
1 vote for fixture, 1 vote for finalizer, 2 abstained from voting... all right, may the `--force` be with us instead of the global safemode setting.